### PR TITLE
3d-tiles: Tileset3D options refactor

### PR DESCRIPTION
**Pull request by ibgreen**
This PR:
- adds `tileset.options` and removes the copying of fields from options object into fields of the `Tileset3D` class.
- removes underscore from some publicly accessible fields, and removes accessors.
- in some cases this makes previously constant fields mutable but it removes a lot of code clutter.

@loshjawrence

### CODE REVIEW

File: modules/3d-tiles/src/tileset/tileset-3d.js [hunk 1]
```diff
@@ -82,15 +84,17 @@ const DEFAULT_OPTIONS = {
...
   onTileUnload: () => {}, // Indicates this a tile's content was unloaded
   onTileLoadFailed: ({tile, message, url}) => {
     // Indicates this a tile's content failed to load
     console.error(`A 3D tile failed to load: ${url} ${message}`); // eslint-disable-line
   },
+
+  // TODO - currently unused callbacks
+  onLoadProgress: () => {}, // Indicates progress of loading new tiles
+  onAllTilesLoaded: () => {}, // All tiles that meet the screen space error of this frame are loaded
+  initialTilesLoaded: () => {}, // Indicates all tiles meet the screen space error this frame loaded
```
Comment (xintongxia) [inline modules/3d-tiles/src/tileset/tileset-3d.js]: https://github.com/uber-web/loaders.gl/pull/326#discussion_r315833869

File: modules/3d-tiles/src/tileset/tileset-3d.js [hunk 2]
```diff
@@ -99,58 +103,75 @@ function getBasePath(url) {
   return slashIndex >= 0 ? url.substr(0, slashIndex) : '';
 }
 
+function getQueryParamString(queryParams) {
```
Comment (xintongxia) [inline modules/3d-tiles/src/tileset/tileset-3d.js]: move to utils or helpers?

File: modules/3d-tiles/src/tileset/tileset-3d.js [hunk 3]
```diff
@@ -161,43 +182,13 @@ export default class Tileset3D {
...
-    this._skipLevelOfDetail = this.skipLevelOfDetail;
-    this._disableSkipLevelOfDetail = false;
-
-    this._gpuMemoryUsageInBytes = 0;
-
     this._initializeTileSet(json, options);
 
-    this.props = {};
-    Object.assign(this.props, options);
+    // Object.freeze(this);
```
Comment (xintongxia) [inline modules/3d-tiles/src/tileset/tileset-3d.js]: remove it?

File: modules/3d-tiles/src/tileset/tileset-3d.js [hunk 4]
```diff
@@ -161,43 +182,13 @@ export default class Tileset3D {
...
-    while (stack.length > 0) {
-      for (const child of tile.children) {
-        stack.push(child);
-      }
-      const tile = stack.pop();
-      tile.destroy();
-    }
-
-    this._root = null;
+    this._destroy();
```
Comment (xintongxia) [inline modules/3d-tiles/src/tileset/tileset-3d.js]: is it necessary to have both `_destroy` and `destroy`?

**Comment by ibgreen** [comment]
@xintongxia Will address nits in next PR.

**Pull request merged**